Remove special handling of libiconv library on Apple platforms to fix:

CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
ICONV_LIBRARIES (ADVANCED)

They already removed the special handling of libiconv headers on Apple platforms years ago:
https://github.com/mariadb-corporation/mariadb-connector-c/commit/a57406cb3b0d4d2153fc2c62c90bb72cbf5b369d

https://github.com/HaxeFoundation/neko/issues/216
--- cmake/patch_mariadb.cmake.orig	2023-01-12 23:26:04.000000000 -0600
+++ cmake/patch_mariadb.cmake	2023-01-12 23:28:31.000000000 -0600
@@ -23,4 +23,17 @@
 	content ${content}
 )
 
+file(WRITE ${cmakelists} ${content})
+
+set(cmakelists ${mariadb_source}/cmake/FindIconv.cmake)
+
+file(READ ${cmakelists} content)
+
+# Don't handle libiconv differently on Apple platforms.
+string(REPLACE
+	"(APPLE)"
+	"(FALSE)"
+	content ${content}
+)
+
 file(WRITE ${cmakelists} ${content})
\ No newline at end of file
